File: 5-6주차/B1931.py
import sys
input = sys.stdin.readline

N = int(input())
meetings = [tuple(map(int, input().split())) for _ in range(N)]
meetings.sort(key=lambda x: (x[1], x[0]))

cnt, end = 0, 0
for s, e in meetings:
    if s >= end:
        end = e
        cnt += 1
print(cnt)

File: 5-6주차/B1931.js
const fs = require('fs');
const input = fs.readFileSync(0, 'utf8').trim().split(/\s+/).map(Number);
const N = input[0];
const arr = [];
for (let i = 1; i < input.length; i += 2) {
  arr.push([input[i], input[i + 1]]);
}

arr.sort((a, b) => (a[1] === b[1] ? a[0] - b[0] : a[1] - b[1]));

let cnt = 0, end = 0;
for (const [s, e] of arr) {
  if (s >= end) {
    end = e;
    cnt++;
  }
}
console.log(cnt);
